src/Renci.SshNet/Sftp/SftpFileStream.cs

```diff
@@ -798,15 +801,16 @@ public override long Seek(long offset, SeekOrigin origin)
                             newPosn = _position + offset;
                             break;
                         case SeekOrigin.End:
-                            var attributes = _session.RequestFStat(_handle, false);
-                            newPosn = attributes.Size - offset;
+                            attributes = _session.RequestFStat(_handle, false);
+                            newPosn = attributes.Size + offset;
                             break;
                     }
 
-                    if (newPosn == -1)
+                    if (newPosn == -1 || (attributes != null && newPosn > attributes.Size))
```

Comment:
seeking past the end of file when writing is valid, it's a way to extend the file size.